config/string_list.go

```diff
@@ -74,3 +74,16 @@ func (sl StringList) String() string {
 func IsStringList(s string) bool {
 	return strings.Contains(s, stringListDelim)
 }
+
+func CompactStringList(sl StringList) StringList {
```

Comment:
If you write this signature as ``func (sl Stringlist) Compact() StringList`` then it can be callable as a method, like ``stringList.Compact()``. Looks like the other ``StringList`` functions further up in this file were defined in this way, so I think it'd be nice for consistency.

(Also looks like those ``StringList`` methods are sorted in alphabetical order, so maybe push this up to its rightful spot in that ordering.)